pkg/flow/flow.go

```diff
@@ -259,14 +277,14 @@ func (f *Flow) Run(ctx context.Context) {
 // file. Components in the graph will be marked as unhealthy if there was an
 // error encountered during Load.
 //
-// The controller will only start running components after Load is called once
-// without any configuration errors.
-func (f *Flow) LoadFile(file *File, args map[string]any) error {
+// If strict is true, the controller will only start running components after
+// Load is called once without any configuration errors.
+func (f *Flow) LoadFile(file *File, args map[string]any, strict bool) error {
 	f.loadMut.Lock()
 	defer f.loadMut.Unlock()
 
 	diags := f.loader.Apply(args, file.Components, file.ConfigBlocks)
-	if !f.loadedOnce.Load() && diags.HasErrors() {
+	if strict && !f.loadedOnce.Load() && diags.HasErrors() {
```

Comment:
The issue here I think is that we allow all the components to use the `strict=false` mode, while it only makes sense for some components.

Could we change the component only - to not return any errors when it encounters a recoverable error and keep retrying inside its `Run`? To not hide the problem, it could report it via `CurrentHealth() Health` but continue running.

Reply:
> Could we change the component only - to not return any errors when it encounters a recoverable error and keep retrying inside its Run? To not hide the problem, it could report it via CurrentHealth() Health but continue running.

I'm firmly against this, because not returning any errors means that you will have downstream evaluation errors for any component which depends on the failed component, even though there is only one actually invalid component.

Reply:
> not returning any errors means that you will have downstream evaluation errors for any component which depends on the failed component, even though there is only one actually invalid component.

If this is combined with making the component that originated the error unhealthy via `CurrentHealth`, then one may be able to see in the UI where the issues start by following the dependency arrows... I get your concern though, having them marked as "unknown" is nicer.

> Can you explain what problem this causes? 

When I set `strict=false` then even unrecoverable errors will allow the graph evaluation to continue, while there may be no way for the component to recover - it may be completely misconfigured or missing some dependency that is not polled. Not every component can recover, in fact most components cannot recover, so I thought it may be too permissive. 

> What does "allow all components to use the strict=false mode" mean for end users?
What I mean is that the setting `strict=false` applies globally. While some components poll and may be able to recover, most components cannot recover.

I am not strongly against the solution in this prototype, but I can see some merits of doing it in the component.

Reply:
thanks for the reply, it makes more sense now